myAPI/API/resources.py

```diff
@@ -0,0 +1,46 @@
+from django.db.models import QuerySet
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+from myAPI.API.serializers import WriterSerializer, BookReadSerializer, BookWriteSerializer, WriterSerializerWithBooksId
+from myAPI.models import Writer, Book
+
+
+class WriterViewSet(ModelViewSet):
+    queryset = Writer.objects.all()
+    serializer_class = WriterSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.request.method == 'GET':
+            book_name = self.request.query_params.get('book_name')
+        if book_name:
```

Comment:
referenced before assignment, if request not get book_name will not exist

Reply:
as I found in docs, we get "query_params" dictionary with all sorts of requests, not only with GET, so deleting line:
      `if self.request.method == 'GET':`
solves the problem and makes code simpler

but if we strongly have to handle only GET - have to assign with None value above IF